staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -744,9 +750,21 @@ func (c *Cacher) List(ctx context.Context, key string, opts storage.ListOptions,
 			return err
 		}
 	}
+	metrics.RecordListCacheMetrics(c.resourcePrefix, indexUsed, len(objs), predicateComplexity(pred), numEvals, listVal.Len())
 	return nil
 }
 
+func predicateComplexity(pred storage.SelectionPredicate) int {
```

Comment:
Going back to the discussion - I'm still not convinced we want that.

I think that knowing if selector was empty vs non-empty is all we need - we won't be able to get any useful information from knowing which exact selector.